diff --git a/backend/backend/settings.py b/backend/backend/settings.py
index 01bbe7d..61f2b67 100644
--- a/backend/backend/settings.py
+++ b/backend/backend/settings.py
@@ -32,10 +32,10 @@ ALLOWED_HOSTS = ['10.0.2.2', 'localhost', 'django-backend', BASE_URL]
 SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
 
 # Firebase settings - Uncomment this block to enable Firebase Cloud Messaging
-# import firebase_admin
-# from firebase_admin import credentials
-# cred = credentials.Certificate("backend/prosumiotest-firebase-adminsdk-9nzkc-13375b0089.json")
-# firebase_admin.initialize_app(cred)
+import firebase_admin
+from firebase_admin import credentials
+cred = credentials.Certificate("backend/prosumiotest-firebase-adminsdk-9nzkc-13375b0089.json")
+firebase_admin.initialize_app(cred)
 
 # Application definition
 
@@ -49,7 +49,7 @@ INSTALLED_APPS = [
     'corsheaders',
     'django_emqx',
     'notifications',
-    # 'fcm_django',  # Uncomment this line to enable Firebase Cloud Messaging
+    'fcm_django',  # Uncomment this line to enable Firebase Cloud Messaging
     'rest_framework',
     'rest_framework.authtoken',
     'rest_framework_simplejwt',
diff --git a/backend/requirements.txt b/backend/requirements.txt
index 2e9f87d..ee2ef14 100644
--- a/backend/requirements.txt
+++ b/backend/requirements.txt
@@ -1,5 +1,5 @@
-git+https://github.com/jakobatgithub/django-emqx.git@main#egg=django-emqx # standard install without local development
-# git+https://github.com/jakobatgithub/django-emqx.git@main#egg=django-emqx[fcm] # standard install without local development and Firebase support
+# git+https://github.com/jakobatgithub/django-emqx.git@main#egg=django-emqx # standard install without local development
+git+https://github.com/jakobatgithub/django-emqx.git@main#egg=django-emqx[fcm] # standard install without local development and Firebase support
 
 # -e django-emqx[dev,test] # For local development
 
